Rumble chat embed

mattroseman opened the pull request:
Rumble recently added popout chats, and even recently-er allowed those popout chats to be embedded on other domains.
So I've added functionality along with YT and Twitch chats.

This implementation is a little more complex, though, because I have to make a request to get the embedded stream's HTML to find the id of that stream's chat.
As far as I can tell there isn't an API to get this information, so I'm downloading the HTML and running a regex search on it.

This is somewhat fragile, and while it should fail gracefully, I would not be surprised if the code becomes outdated as the embedded stream's HTML changes.
Up to you if you want to add it :)

mattroseman committed:
added ability to view embedded Rumble stream's chat

mattroseman committed:
added error handling for getting rumble chat id

vyneer commented:
Heyo! Thanks for the PR, this seems like a good thing to add. Parsing the page for the ID _is_ a little clunky and, luckily, I know a better way to get it (haven't tested it much, but I don't see any reason it shouldn't work all the time): the numerical ID is just the embed ID (without the 'v') converted from 36 base to decimal 😄 
So, for example, for embed ID 'v2an734', where chat popup URL is 'https://rumble.com/chat/popup/138810784' (just picked some random rumble stream dont judge me PepeLaugh), you can just do ```parseInt('2an734', 36)```, which gets you 138810784 😎 
Once you change that up I'll merge ofc!
Thanks again ❤️

mattroseman committed:
get Rumble stream's chat id directly from embed id

Instead of making an additional query for the Rumble stream's chat id,
convert the base36 embed id to the base10 chat id.

mattroseman commented:
That's an impressive find; I just assumed those were different ids, not different bases of the same number 😮 
That makes this a lot easier, and basically the same flow as YT and Twitch. Thanks for the help :heart:

I've updated the PR to just convert the embed id to the chat id, and there's no more additional query.

vyneer commented:
Yeah, I spent a day trying to figure out how they store their stuff after that Sneako stream that Steve was on got privated :smile: You can also get some other ID from the regular URL as well, which you can convert to some base64-ish string that you can use to get the mp4 files from their CDN. Made a lidl Go program for it, should prob make a repo for it.

Anyway, this looks good! Thanks for the PR as always :heart:
